mesonbuild/compilers.py

```diff
@@ -2317,11 +2286,19 @@ def get_warn_args(self, level):
 
 
 class GnuFortranCompiler(FortranCompiler):
-    def __init__(self, exelist, version, gcc_type, is_cross, exe_wrapper=None):
+    def __init__(self, exelist, version, gcc_type, is_cross, exe_wrapper=None, defines=None):
         super().__init__(exelist, version, is_cross, exe_wrapper=None)
         self.gcc_type = gcc_type
+        self.defines = defines or {}
         self.id = 'gcc'
 
+    def has_define(self, define):
+        return define in self.defines
+
+    def get_define(self, define):
+        if define in self.defines:
```

Comment:
I benchmarked this a long time ago. This is faster than `.get()` in every case, and is much faster in the negative case (when define not in defines). IIRC, it's because `.get()` sets up the try...except machinery.